libs/@blockprotocol/graph/src/codegen/context/postprocess.ts

```diff
@@ -0,0 +1,205 @@
+  readonly urlsToCompiledTypes: Record<
+    keyof typeof this.allTypes,
+    CompiledTsType
+  >;
+
+  /**
+   * Map of TypeScript identifiers, to their source (the file that exports them) and the contents required to define
+   * them if they're a locally defined type
+   */
+  IdentifiersToSources: Record<string, IdentifierSources>;
+
+  /** Map of files to the identifiers of types that they will **depend on**, either by defining or importing them */
+  filesToDependentIdentifiers: Record<string, Set<string>> = {};
+
+  /** Map of files to the identifiers of types that they will **define** */
+  filesToDefinedIdentifiers: Record<string, Set<string>> = {};
+
+  /** Map of files to the actual TypeScript contents */
+  filesToContents: Record<string, string> = {};
+
+  constructor(compileContext: CompileContext) {
+    this.parameters = compileContext.parameters;
+    this.logLevel = compileContext.logLevel;
+    this.dataTypes = compileContext.dataTypes;
+    this.propertyTypes = compileContext.propertyTypes;
+    this.entityTypes = compileContext.entityTypes;
+    this.allTypes = compileContext.allTypes;
+    this.typeDependencyMap = compileContext.typeDependencyMap;
+    this.linkTypeMap = compileContext.linkTypeMap;
+    this.urlsToCompiledTypes = compileContext.urlsToCompiledTypes;
+
+    const graphModuleImportPath = this.parameters.temporal
+      ? "@blockprotocol/graph/temporal"
+      : "@blockprotocol/graph";
+
+    this.IdentifiersToSources = {
+      Entity: {
+        locallyImportable: true,
+        source: { kind: "external", definingPath: graphModuleImportPath },
+      },
+      LinkData: {
+        locallyImportable: true,
+        source: { kind: "external", definingPath: graphModuleImportPath },
+      },
+      JsonObject: {
+        locallyImportable: true,
+        source: { kind: "external", definingPath: graphModuleImportPath },
+      },
+      LinkAndRightEntity: {
+        locallyImportable: true,
+        source: {
+          kind: "external",
+          definingPath: graphModuleImportPath,
+        },
+      },
+    };
+  }
+
+  /* @todo - Replace this with a proper logging implementation */
+  logWarn(message: string) {
+    if (this.logLevel !== "silent") {
+      // eslint-disable-next-line no-console
+      console.warn(`WARN: ${message}`);
+    }
+  }
+
+  logInfo(message: string) {
+    if (
+      this.logLevel === "info" ||
+      this.logLevel === "debug" ||
+      this.logLevel === "trace"
+    ) {
+      // eslint-disable-next-line no-console
+      console.log(`INFO: ${message}`);
+    }
+  }
+
+  logDebug(message: string) {
+    if (this.logLevel === "debug" || this.logLevel === "trace") {
+      // eslint-disable-next-line no-console
+      console.log(`DEBUG: ${message}`);
+    }
+  }
+
+  logTrace(message: string) {
+    if (this.logLevel === "trace") {
+      // eslint-disable-next-line no-console
+      console.log(`TRACE: ${message}`);
+    }
+  }
```

Comment:
This is shared between the different contexts. Do we want to centralize this along with the todo at the top?

Reply:
The actual implementations are a line or two so I don't see turning them into functions as helpful. The other way I could see us centralising it is by making some context that all the others `extend` but that seems kind of superfluous..

Really the solution is to use `winston` or something for logging, in which case you generally need a `logger` object to pass around which is why I put the methods in the classes for now as opposed to some standalone thing.